### ISSUE TITLE
 `DatePicker`: Add `on-panel-change` Support to `DatePicker`, Similar to `Calendar`

### ISSUE THREAD

**Issue by SK-ERIC** [tusen-ai/naive-ui#5783]
### Clear and concise description of the problem

I am experiencing an issue that seems to be related to a previously reported problem (see [#5350](https://github.com/tusen-ai/naive-ui/issues/5350)). However, the current solutions provided do not adequately address my needs for `DatePicker` panel switches.

In the current implementation, the methods `on-next-month` and `on-next-year` do not facilitate monitoring of `DatePicker` panel transitions. This limitation becomes apparent when trying to track the user's interaction with the panel, specifically when they switch between months or years using the provided `Select` component within the panel.### 

### Suggested solution

Ideally, I would like to have a way to monitor and respond to the panel switching events, especially when users navigate through different months and years using the `Select` component. This feature would significantly enhance the usability and interactivity of the `DatePicker` panel.

### Alternative

Additionally, I would like to highlight an existing feature in the `Calendar` component that aligns with the functionality I am seeking. The `Calendar` component provides an `on-panel-change` event ([NCalendar API](https://www.naiveui.com/en-US/os-theme/components/calendar#API)) that allows monitoring of changes in the panel, such as switching between months or years.

### Additional context

![image](https://github.com/tusen-ai/naive-ui/assets/52649359/af9f3ec1-fb90-4f69-8ed6-ba1afb231ed4)

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/tusen-ai/naive-ui/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://www.naiveui.com/en-US).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by jahnli** [comment]
`on-update:show`   ?

**Comment by SK-ERIC** [comment]
> `on-update:show` ?

No, what I meant is not to monitor the opening or closing state of the time panel. I am referring to the changes in the dates on the panel, which corresponds to the on-panel-change event of the Calendar component. My focus is on the callback behavior when there is a change in the time panel. Therefore, the required event is not on-update:show, but rather on-panel-change.

![image](https://github.com/tusen-ai/naive-ui/assets/52649359/da97af60-f98d-4148-a4db-413d02102824)

**Comment by jahnli** [comment]
![image](https://github.com/user-attachments/assets/ecd825de-d6e2-453d-984b-f3f92542333f)

**Comment by llifangyin** [comment]
并没有解决啊，我们想要点击日期面板就触发的事件。
比如日期范围选择器，选择第一个日期就触发的事件，不用选择完或者点击确认按钮就能触发的事件；

**Comment by jahnli** [comment]
可以重开一个 issue 详细描述

**Comment by jahnli** [comment]
https://github.com/tusen-ai/naive-ui/issues/6459
